File: repeatedString.js

// console.log(count)
// Complete the repeatedString function below.
function repeatedString(s, n) {
    x = s.length
    a = []
    j = 0;
    for (i = 0; i < n; i) {
        if (j < x) {
            a.push(s[j])
            j++;
            i++;
        } else {
            j = 0;
        }

    }
    count = 0;
    for (let j = 0; j < n; j++) {
        if (a[j] === 'a') {
            count++;
        }
    }
    // let lenS = s.length;
    // let i = 0;
    // let charArr = [];
    // for (let j = 0; j < lenS; j) {
    //     if (i < lenS) {
    //         charArr.push(s[i]);
    //         i++;
    //         j++;
    //     } else {
    //         i = 0;
    //     }
    // }
    // let count = 0;
    // for (let j = 0; j < n; j++) {
    //     if (charArr[j] === 'a') {
    //         count++;
    //     }
    // }
    return count
}
console.log(repeatedString('aba', 10))

File: repeatedString.py
def repeatedString(s, n):
    # x = len(s)
    # a = []
    # j = 0
    # i = 0
    # while i < n:
    #     if j < x:
    #         a.append(s[j])
    #         j += 1
    #         i += 1
    #     else:
    #         j = 0
    # count = 0
    # for i in range(n):
    #     if a[i] == 'a':
    #         count += 1
    # print(a)
    # return count
    return s.count("a") * (n // len(s)) + s[:n % len(s)].count("a")


print(repeatedString('aba', 10))
